src/nft/components/sell/select/SelectPage.tsx

```diff
@@ -1,3 +1,380 @@
+    }
+  )
+
+  const ownerCollectionsAddresses = useMemo(() => ownerCollections?.map(({ address }) => address), [ownerCollections])
+  const { data: collectionStats } = useQuery(
+    ['ownerCollectionStats', ownerCollectionsAddresses],
+    () => fetchMultipleCollectionStats({ addresses: ownerCollectionsAddresses ?? [] }),
+    {
+      refetchOnWindowFocus: false,
+    }
+  )
+
+  const {
+    data: ownerAssetsData,
+    fetchNextPage,
+    hasNextPage,
+    isSuccess,
+  } = useInfiniteQuery(
+    ['ownerAssets', address, collectionFilters],
+    async ({ pageParam = 0 }) => {
+      return await fetchWalletAssets({
+        ownerAddress: address ?? '',
+        collectionAddresses: collectionFilters,
+        pageParam,
+      })
+    },
+    {
+      getNextPageParam: (lastPage, pages) => {
+        return lastPage?.flat().length === 25 ? pages.length : null
+      },
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+    }
+  )
+
+  const ownerAssets = useMemo(() => (isSuccess ? ownerAssetsData?.pages.flat() : null), [isSuccess, ownerAssetsData])
+
+  const walletAssets = useWalletCollections((state) => state.walletAssets)
+  const setWalletAssets = useWalletCollections((state) => state.setWalletAssets)
+  const displayAssets = useWalletCollections((state) => state.displayAssets)
+  const setDisplayAssets = useWalletCollections((state) => state.setDisplayAssets)
+  const setWalletCollections = useWalletCollections((state) => state.setWalletCollections)
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const reset = useSellAsset((state) => state.reset)
+  const setSellPageState = useSellPageState((state) => state.setSellPageState)
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    setWalletAssets(ownerAssets?.flat() ?? [])
+  }, [ownerAssets, setWalletAssets])
+
+  useEffect(() => {
+    ownerCollections && setWalletCollections(ownerCollections)
+  }, [ownerCollections, setWalletCollections])
+
+  const listFilter = useWalletCollections((state) => state.listFilter)
+
+  useEffect(() => {
+    if (searchText) {
+      const filtered = walletAssets.filter((asset) => asset.name?.toLowerCase().includes(searchText.toLowerCase()))
+      setDisplayAssets(filtered, listFilter)
+    } else {
+      setDisplayAssets(walletAssets, listFilter)
+    }
+  }, [searchText, walletAssets, listFilter, setDisplayAssets])
+
+  useEffect(() => {
+    setDisplayAssets(walletAssets, listFilter)
+  }, [walletAssets, listFilter, setDisplayAssets])
+
+  useEffect(() => {
+    if (ownerCollections?.length && collectionStats?.length) {
+      const ownerCollectionsCopy = [...ownerCollections]
+      for (const collection of ownerCollectionsCopy) {
+        const floorPrice = collectionStats.find((stat) => stat.address === collection.address)?.floorPrice
+        collection.floorPrice = floorPrice ? Math.round(floorPrice * 1000 + Number.EPSILON) / 1000 : 0 //round to at most 3 digits
```

Comment:
nit: would suggest making a helpful function for this so the comment isn't necessary